### ISSUE TITLE
Some YUV sample videos are tagged in the wrong color range

### ISSUE THREAD

**Issue by YomikoR** [jursonovicst/gradient#8]
Hello,

I think the test samples `\test_sequences\1920x1080\gradient_1920-1080_0-25_gbrp10le_x265.mp4` and `\test_sequences\1920x1080\gradient_1920-1080_0-25_yuv444p10_x265.mp4` are supposed to be showing the same colors, but they're not.

Based on the actual pixel data I guess the reason is as follows: the YUV colors were converted from the RGB video, namely, from full-range RGB to limited-range YUV. Then the YUV video was tagged as full range by the encoding toolchain (the encoder usually doesn't take conversion, just tags).

Since most video players capable of handling 10-bit output are using this tag, it deserves to be fixed.

Thanks

**Comment by ValZapod** [comment]
Full range has 0, 1, 2, 3 values and 1023, 1022, 1021, 1020. Those are reserved in limited, so...

**Comment by damian101** [comment]
Yes, all the yuv videos are wrongly tagged as full-range. Causes problems in video players respecting the tag, like mpv.

FFmpeg by default uses limited range when doing conversion to yuv, the full-range option in x265 does not change encoding, only adds the full-range tag.

**Comment by damian101** [comment]
> That is the default, so you do not need to tell it anything. And no, HD material will also use BT.601. Same about decoding if not tagged.
> 
> https://trac.ffmpeg.org/wiki/Scaling "The default for matrix in untagged input and output is always limited BT.601" Old JPEG is tagged output, it is always full range BT.601. Jpeg2000 can have BT.709 matrix too though.
> 
> And BTW real color space is primaries, not matrix. Matrix was never supposed to be something you can mix up.

Oh, that's what you mean. Yes, if source is untagged FFmpeg assumes BT.601. Otherwise the color space tag is just carried over. Color space conversion still never happens by default.
